app/main.py

```diff
@@ -2,0 +1,55 @@
+from time import sleep 
+
+
+class SoftwareEngineer:
+    def __init__(self, name: str, skills: list) -> None:
+        self.name = name
+        self.skills = skills
+    
+    def learn_skill(self, skill: str) -> None:
+        self.__class__.skills.append(skill)
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name, skills):
+        super().__init__(name, skills)
+    
+    skills = ["JavaScript", "HTML", "CSS"]
+    
+    def create_awesome_web_page(self):
+        print(f"{self.name} is creating a webpage...")
+        sleep(2)
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name, skills):
+        super().__init__(name, skills)
+    
+    skills = ["Python", "SQL", "Django",]
+    
+    def create_awesome_web_page(self):
+        print(f"{self.name} is creating an API...")
+        sleep(2)
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(self, name, skills):
+        super().__init__(name, skills)
+    
+    skills = ["Java", "Android studio"]
```

Comment:
The `skills` attribute is defined at the class level, causing it to be shared across all instances. Move it to the `__init__` method to ensure each instance has its own list of skills.